gson/src/org/immutables/gson/stream/JsonParserReader.java

```diff
@@ -141,7 +144,7 @@ public String nextName() throws IOException {
   public String nextString() throws IOException {
     requirePeek();
     if (!isLenient()) {
-      expect(VALUE_STRING);
+      expectOneOf(VALUE_STRING, VALUE_NUMBER_FLOAT, VALUE_NUMBER_INT);
```

Comment:
Major question, maybe we just need to enable lenient mode? Here, `!isLenient()` means "strict mode", in which it would be natural to expect only strings. By accepting `float` and `int` as string means that the mode is no longer "strict" mode. Are there any problems to enable lenient mode?